anet-tools/src/examples/models.py

```diff
@@ -0,0 +1,215 @@
+import copy
+import datetime
+import uuid
+
+from sqlalchemy import Boolean, Column, DateTime, Float, ForeignKey, Integer, LargeBinary, String, Text, text
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+from sqlalchemy_mixins import ActiveRecordMixin
+
+from src.core.base_methods import base_methods
+
+
+Base = declarative_base()
+metadata = Base.metadata
+
+
+class BaseModel(Base, ActiveRecordMixin):
+    __abstract__ = True
+
+
+class anet_logic_mixin(BaseModel):
+    __abstract__ = True
+
+    def insert_entity(self, createdAt):
+        """Insert and flush a new record
+        """
+        self.createdAt = createdAt
+        self.updatedAt = createdAt
+        if self.__tablename__ == "people":
+            utc_now = datetime.datetime.now()
+            PeoplePositions.create(createdAt=utc_now, person=self)
+        else:
+            BaseModel.session.add(self)
+            BaseModel.session.flush()
+
+    def update_entity(self, updatedAt):
+        """Update and flush an existing record
+        """
+        obj = type(self).find(self.uuid)
+
+        self.updatedAt = updatedAt
+        for attr, value in self.__dict__.items():
+            if attr != "_sa_instance_state":
+                setattr(obj, attr, value)
+        BaseModel.session.flush()
+
+    def insert_update_nested_entity(self, update_rules, utc_now):
+        self_c = copy.deepcopy(self)
+
+        if base_methods.has_entity_relation(self, "person"):
```

Comment:
Handling people specific tasks would belong in the sub-class people. Same for location, organisation & person.